controllers/result.go

```diff
@@ -77,6 +80,52 @@ func (c *ResultController) Create() {
 	c.Redirect("/", http.StatusFound)
 }
 
+func (c *ResultController) Show() {
+	c.EnsureAuthenticatedUser()
+	c.TplName = "results/show.html"
+	c.Data["Title"] = "Result Detail"
+	web.ReadFromRequest(&c.Controller)
+
+	resultID, err := c.getResultID()
+	if err == nil {
+		result, err := models.GetResultByIdWithRelations(resultID)
+		if err != nil {
+			logs.Error("Failed to get result:", err.Error())
+		}
+
+		c.Data["result"] = result
+	}
+}
+
+func (c *ResultController) Cache() {
+	c.EnsureAuthenticatedUser()
+	c.TplName = "results/cache.html"
+	c.Data["Title"] = "Result Page Cache"
+	web.ReadFromRequest(&c.Controller)
+
+	resultID, err := c.getResultID()
+	if err == nil {
```

Comment:
Possible return early in case the error is present? then 2 level nested blocks will be flattened also I think. What do you think?

Reply:
Fixed on [3e5e714](https://github.com/carryall/go-google-scraper-challenge/pull/72/commits/3e5e714ac9ac92748a798395d061496ae05ba81a)